testsys/src/delete.rs

```diff
@@ -36,15 +43,81 @@ derive_fromstr_from_deserialize!(ObjectType);
 
 impl Delete {
     pub(crate) async fn run(&self, k8s_client: Client) -> Result<()> {
-        match self.object_type {
-            ObjectType::Test => {
-                delete_test(k8s_client, &self.object_name, self.include_resources).await
+        match (self.all, self.object_type, self.object_name.as_ref()) {
+            (true, _, _) => delete_all(k8s_client).await,
+            (false, Some(ObjectType::Test), Some(object_name)) => {
+                delete_test(k8s_client, object_name, self.include_resources).await
+            }
+            (false, Some(ObjectType::Resource), Some(object_name)) => {
+                delete_resource(k8s_client, object_name, self.include_resources).await
             }
-            ObjectType::Resource => {
-                delete_resource(k8s_client, &self.object_name, self.include_resources).await
+            (_, _, _) => Err(error::Error::InvalidArguments {
+                why: "Either `all` must be set, or both of `object-type` and `object-name`"
+                    .to_string(),
+            }),
+        }
+    }
+}
+
+async fn delete_all(k8s_client: Client) -> Result<()> {
+    let test_client = TestClient::new_from_k8s_client(k8s_client.clone());
+    let resource_client = ResourceClient::new_from_k8s_client(k8s_client.clone());
+
+    // Start by deleting all tests and waiting for their completion.
+    println!("Deleting all tests...");
+    test_client.delete_all().await.context(error::DeleteSnafu {
+        what: "all tests".to_string(),
+    })?;
+    while !test_client
+        .get_all()
+        .await
+        .context(error::GetSnafu { what: "all tests" })?
+        .is_empty()
+    {
+        tokio::time::sleep(Duration::from_secs(5)).await;
+    }
```

Comment:
We should move the `wait` procedures out and call them with timeouts. I'm not sure how long to wait, a few minutes?

Reply:
I am worried about setting timeouts because deleting a cluster takes forever.

Reply:
Makes sense.